### PR TITLE
Internal: Replaced dev-dependencies with goalgoriall/open_social_dev.

### PR DESCRIPTION
**Pull request by navneet0693**
## Problem
Currently, we are maintaining dev/test dependencies at different places.

https://github.com/goalgorilla/open_social/blob/main/composer.json#L224
https://github.com/goalgorilla/open_social/blob/main/tests/composer.json#L9
https://github.com/goalgorilla/open_social_docker/blob/master/composer.json#L30
https://github.com/goalgorilla/drupal_social/blob/master/composer.json#L30

This made it difficult to update the dependencies and also to maintain them in different places. The dependencies version is the same across our repositories. This lead to a lot of time consumption of doing the same thing in different places, also because these repositories are coupled tightly, outdated repository lead to the failure of others.

## Solution
Our inspiration comes from https://packagist.org/packages/drupal/core-dev .

See: https://www.drupal.org/docs/develop/using-composer/starting-a-site-using-drupal-composer-project-templates#s-adding-drupalcore-dev

The new repository "https://github.com/goalgorilla/open_social_dev" contains all these dependencies required for development and testing.

1. Use the created repository to replace the dev dependencies.
2. Update tests/compose.json to require this repo.
3. Remove the dependabot action to update the dev repository as it is already added in the created repo.

## Issue tracker
N.A

## Theme issue tracker
N.A

## How to test
- [ ] All checks should pass.

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N.A

## Release notes
N.A

## Change Record
N.A

## Translations
N.A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2992-olspvhbqphxp7t1x9c05nnh0s5ebjlsu.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62ab2677232f1eb8e76240bb

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2992-olspvhbqphxp7t1x9c05nnh0s5ebjlsu.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62ab2677232f1eb8e76240bb

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2992-olspvhbqphxp7t1x9c05nnh0s5ebjlsu.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62ab2677232f1eb8e76240bb

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2992-olspvhbqphxp7t1x9c05nnh0s5ebjlsu.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62ab2677232f1eb8e76240bb

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2992-olspvhbqphxp7t1x9c05nnh0s5ebjlsu.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62ab2677232f1eb8e76240bb

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2992-olspvhbqphxp7t1x9c05nnh0s5ebjlsu.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62ab2677232f1eb8e76240bb

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2992-olspvhbqphxp7t1x9c05nnh0s5ebjlsu.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62ab2677232f1eb8e76240bb

**Review by Kingdutch — COMMENTED** [review]
Fan of this change! Though a few questions/remarks.

What is the reason we're overhauling our GitHub workflows in the same PR and also changing our test configs as a result? e.g. changing our `phpcs.xml` could break local developer workflows (since it changes how PHPCS is called), same for PHPStan (and not PHPUnit?)

I would expect this change to be mostly a drop-in replacement of our current dev dependencies to use the new package with versions. That ensures less risk of breaking people's workflow (or a smaller change to undo, if one of our steps breaks a workflow).

My proposal would be to move the changes to our test workflows to a separate PR. That allows this one with the dev-dependencies package to be merged quickly, improving maintenance, while bikeshedding happens over the testing set-up :) 

Additionally I'm wondering if we should start out with a strict version of the dev dependencies immediately. Using `dev-main` could cause tests on a prior version branch to start failing because we need newer tooling for our main branch (e.g. due to PHPUnit compatibility). If we decide not to version immediately it's probably a good idea to document why :D

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -10199,6 +10199,16 @@ parameters:
 			count: 2
 			path: modules/social_features/social_group/modules/social_group_invite/social_group_invite.module
 
+		-
```
Comment (bojanbogdanovic) [inline phpstan-baseline.neon:10202]: Did you try to resolve these phpstan issues in code? We should try to avoid adding such exceptions in the baseline.

**Comment by navneet0693** [comment]
> Fan of this change! Though a few questions/remarks.
> 
> What is the reason we're overhauling our GitHub workflows in the same PR and also changing our test configs as a result? e.g. changing our `phpcs.xml` could break local developer workflows (since it changes how PHPCS is called), same for PHPStan (and not PHPUnit?)
> 
> I would expect this change to be mostly a drop-in replacement of our current dev dependencies to use the new package with versions. That ensures less risk of breaking people's workflow (or a smaller change to undo, if one of our steps breaks a workflow).
> 
> My proposal would be to move the changes to our test workflows to a separate PR. That allows this one with the dev-dependencies package to be merged quickly, improving maintenance, while bikeshedding happens over the testing set-up :)
> 
> Additionally I'm wondering if we should start out with a strict version of the dev dependencies immediately. Using `dev-main` could cause tests on a prior version branch to start failing because we need newer tooling for our main branch (e.g. due to PHPUnit compatibility). If we decide not to version immediately it's probably a good idea to document why :D

Q. Why are we changing the workflow?
A. There are two `composer.json` file being used in our tests. One is https://github.com/goalgorilla/open_social/blob/main/composer.json and https://github.com/goalgorilla/open_social/blob/main/tests/composer.json. These both require-dev dependencies. Replacing them simply won't achieve our goals to minimize the usage of dev-dependencies in different places. Also, PHPCS and PHPStan workflow isn't accurately replicated on developer's local as in developer's local environment, they won't do a composer install inside `social` directory.

Q. Why is PHPunit workflow not changed?
A. The way we had setup our PHPCS and PHPStan changes were to use dev-dependencies from https://github.com/goalgorilla/open_social/blob/main/composer.json. And, PHPUnit setup was using https://github.com/goalgorilla/open_social/blob/main/tests/composer.json. In current PR, I created a common project setup using the same method as our PHPUnit test is following for both PHP 7 and PHP 8.

Q. Why are we changing our test configs as a result?
A. Now, because our workflow is setting up project very similar to https://github.com/goalgorilla/social_template or https://github.com/goalgorilla/drupal_social, the scans are running from "root" directory. Therefore, we have updated paths in `phpstan.neon` file and `phpcs.xml` file.

Q.  Will this break developer's local workflow ?
A. No, it won't. The developer has to call the executables in the same form as tests are calling.

Q. Can we use strict versions?
A. The current plan was to copy dependencies as it is without any changes and come up with a plan later to take inspiration from drupal/core-dev to ensure we are following versioning in sync with Open Social.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2992-olspvhbqphxp7t1x9c05nnh0s5ebjlsu.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62ab2677232f1eb8e76240bb

File: phpstan-baseline.neon (hunk 1 above)
Comment (navneet0693) [reply phpstan-baseline.neon]: @bojanbogdanovic Resolved ;-)

**Comment by navneet0693** [comment]
Improvements are added in https://github.com/goalgorilla/open_social/pull/3007

**Review by tbsiqueira — APPROVED** [review]

**Pull request merged**